### PR TITLE
Bootstrap command refactoring: quality-of-life improvements (step 4)

### PR DESCRIPTION
**Pull request by Kobzol**
Continuation of https://github.com/rust-lang/rust/pull/127120.

This PR simply introduce two new functions (`BootstrapCommand:run` and `command`) that make it a bit easier to use commands in bootstrap. It also adds several `#[must_use]` annotations. This shouldn't (hopefully) have any effect on behavior.

Especially the first commit IMO makes any code that runs commands more readable, and allows using the API in a fluent way, without needing to jump back and forth between the command and the `Build(er)`.

Tracking issue: https://github.com/rust-lang/rust/issues/126819

r? @onur-ozkan

### CODE REVIEW

File: src/bootstrap/src/utils/helpers.rs [hunk 1]
```diff
@@ -502,7 +502,7 @@ pub fn check_cfg_arg(name: &str, values: Option<&[&str]>) -> String {
 /// `BootstrapCommand::new("git")`, which is painful to ensure that the required change is applied
 /// on each one of them correctly.
 pub fn git(source_dir: Option<&Path>) -> BootstrapCommand {
-    let mut git = BootstrapCommand::new("git");
+    let mut git = command("git");
```
Comment (onur-ozkan) [inline src/bootstrap/src/utils/helpers.rs:505]: Comment seems outdated

File: src/bootstrap/src/core/build_steps/dist.rs [hunk 2]
```diff
@@ -1703,168 +1703,159 @@ impl Step for Extended {
...
+                    .arg("RustAnalyzerGroup")
                     .arg("-dr")
-                    .arg("Cargo")
+                    .arg("RustAnalyzer")
                     .arg("-var")
-                    .arg("var.CargoDir")
+                    .arg("var.RustAnalyzerDir")
                     .arg("-out")
-                    .arg(exe.join("CargoGroup.wxs"))
+                    .arg(exe.join("RustAnalyzerGroup.wxs"))
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/build_steps/dist.rs:1781]: I assume this was unintentional?

Comment (Kobzol) [reply src/bootstrap/src/core/build_steps/dist.rs:1781]: Sorry, I forgot to add a note here. If you take a closer look, the code actually didn't change at all, but Git(Hub) renders the diff in a terrible way.

File: src/bootstrap/src/utils/helpers.rs (hunk 1 above)
Comment (Kobzol) [reply src/bootstrap/src/utils/helpers.rs]: Good catch, fixed the comment.

**Review by onur-ozkan — APPROVED** [review]

**Pull request merged**